lib/compat.php

```diff
@@ -228,6 +228,39 @@ function gutenberg_get_post_type_capabilities( $user, $name, $request ) {
 	return $value;
 }
 
+/**
+ * Make sure oEmbed REST Requests apply the WP Embed security mechanism for WordPress embeds.
+ *
+ * @see  https://core.trac.wordpress.org/ticket/32522
+ *
+ * TODO: This is a temporary solution. Next step would be to edit the WP_oEmbed_Controller,
+ * once merged into Core.
+ *
+ * @since 2.1.0
+ *
+ * @param  WP_HTTP_Response|array|WP_Error $response The REST Request response.
+ * @param  WP_REST_Server                  $handler  ResponseHandler instance (usually WP_REST_Server).
+ * @param  WP_REST_Request                 $request  Request used to generate the response.
+ * @return WP_HTTP_Response|array                    The REST Request response.
+ */
+function gutenberg_filter_oembed_result( $response, $handler, $request ) {
+	if ( is_wp_error( $response ) || 'GET' !== $request->get_method() ) {
+		return $response;
+	}
+
+	// External embeds.
+	if ( '/oembed/1.0/proxy' === $request->get_route() ) {
+		$response->html = apply_filters( 'oembed_result', _wp_oembed_get_object()->data2html( $response, $_GET['url'] ), $_GET['url'], array() );
+
+		// Internal embeds.
+	} elseif ( '/oembed/1.0/embed' === $request->get_route() ) {
+		$response['html'] = apply_filters( 'oembed_result', _wp_oembed_get_object()->data2html( (object) $response, $_GET['url'] ), $_GET['url'], array() );
```

Comment:
This is removing the `<script>` from the public embed response, which is necessary for non-WordPress sites to be able to handle WordPress embeds. We shouldn't be filtering the response for this endpoint.